src/aiodynamo/sign.py

```diff
@@ -0,0 +1,116 @@
+import datetime
+import hashlib
+import hmac
+import json
+from dataclasses import dataclass
+from typing import *
+
+from yarl import URL
+
+from .credentials import Key
+
+SERVICE = "dynamodb"
+CONTENT_TYPE = "application/x-amz-json-1.0"
+API_VERSION = "DynamoDB_20120810"
+METHOD = "POST"
+ALGORITHM = "AWS4-HMAC-SHA256"
+
+
+def make_default_endpoint(region: str) -> URL:
+    return URL.build(scheme="https", host=f"{SERVICE}.{region}.amazonaws.com", path="/")
+
+
+@dataclass(frozen=True)
+class Instant:
+    _dt: datetime.datetime
+
+    @classmethod
+    def now(cls):
+        return cls(datetime.datetime.now(datetime.timezone.utc))
+
+    @property
+    def timestamp(self) -> str:
+        return self._dt.strftime("%Y%m%dT%H%M%SZ")
+
+    @property
+    def date(self) -> str:
+        return self._dt.strftime("%Y%m%d")
+
+
+@dataclass(frozen=True)
+class Request:
+    url: URL
+    headers: Dict[str, str]
+    body: bytes
+
+
+def derive_signing_key(key: Key, instant: Instant, region: str) -> bytes:
+    tmp_key1 = sha256_hmac(("AWS4" + key.secret).encode("utf-8"), instant.date)
+    tmp_key2 = sha256_hmac(tmp_key1, region)
+    tmp_key3 = sha256_hmac(tmp_key2, SERVICE)
+    return sha256_hmac(tmp_key3, "aws4_request")
+
+
+def signed_dynamo_request(
```

Comment:
Would be nice to benchmark this, I'm sure it's way faster than custom-weird-json-encode-decode, but it would be nice to have some idea what share of the runtime accounts for signing and possible unsigning...

Reply:
pyperf tells me `Benchmark hidden because not significant (1): sign`